t/router/radixtree-sni2.t

```diff
@@ -543,3 +543,85 @@ qr/(fetch|release) table \w+/
 --- grep_error_log_out
 fetch table api_ctx
 release table api_ctx
+
+
+
+=== TEST 15: store secret into vault
+--- exec
+VAULT_TOKEN='root' VAULT_ADDR='http://0.0.0.0:8200' vault kv put kv/apisix/ssl test2.com.crt=@t/certs/test2.crt test2.com.key=@t/certs/test2.key
+--- response_body
+Success! Data written to: kv/apisix/ssl
+
+
+
+=== TEST 16: set ssl conf with secret ref
+--- request
+GET /t
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            -- put secret vault config
+            local code, body = t('/apisix/admin/secrets/vault/test1',
+                ngx.HTTP_PUT,
+                [[{
+                    "uri": "http://127.0.0.1:8200",
+                    "prefix": "kv/apisix",
+                    "token" : "root"
+                }]]
+                )
+            if code >= 300 then
+                ngx.status = code
+                return ngx.say(body)
+            end
+            -- set ssl
+            local code, body = t('/apisix/admin/ssls/1',
+                ngx.HTTP_PUT,
+                [[{
+                    "cert": "$secret://vault/test1/ssl/test2.com.crt",
+                    "key": "$secret://vault/test1/ssl/test2.com.key",
+                    "sni": "test2.com"
+                }]]
+            )
+            if code >= 300 then
+                ngx.status = code
+                return ngx.say(body)
+            end
+
+            ngx.say("passed")
+        }
+    }
+--- response_body
+passed
+
+
+
+=== TEST 17: get cert from vault
```

Comment:
could we add test cases about using `environment variable`?

Reply:
done